src/cursor.rs

```diff
@@ -0,0 +1,171 @@
+use core::fmt;
+use std::{error::Error, sync::Arc};
+
+use crate::{icon::PIXEL_SIZE, platform_impl::PlatformCustomCursor};
+
+#[derive(Debug, Clone)]
+pub struct CustomCursor {
+    pub(crate) inner: Arc<PlatformCustomCursor>,
+}
+
+impl PartialEq for CustomCursor {
+    fn eq(&self, other: &Self) -> bool {
+        Arc::ptr_eq(&self.inner, &other.inner)
+    }
+}
+
+impl Eq for CustomCursor {}
```

Comment:
That can be added by the `derive` macro, no?

Reply:
PartialEq is special because it compares pointers so I don't think so. But now I noticed that I don't even ever compare these, I probably had some idea that didn't work. Do we even want Eq?

Reply:
We do, it can be cloned and is user facing, so definitely helpful.
`PartialEq` needs a manual implementation yes, but `Eq` doesn't.

Reply:
Implementing `PartialEq` manually and `Eq` with `derive` together doesn't build. So it seems we need that impl.

Reply:
Ah, I see, because it does some internal asserts over `PlatformCustomCursor`.
Alright then!